### PR TITLE
(case 20096) Fix missing support for UV scaling parameters in FBX files

### PR DESCRIPTION
**Pull request by sabrina-shanman**
https://highfidelity.manuscript.com/f/cases/20096/UV-scaling-and-tiling-from-FBX-not-being-respected-by-HF

This PR fixes a bug where a model does not preserve the scale of its texture when the model is exported to an FBX file from Maya and then imported into Interface.

It adds support for two different ways of handling uv scale/translation, one of which was partially implemented but didn't work. It also adjusts the way these parameters are handled.

## TEST PLAN
- Run this test comparing texture scaling on three cubes:
    - https://github.com/sabrina-shanman/hifi_tests/tree/fbx_uv-scale/tests/content/entity/model/modelReaders/fbxReader/uv_scale
    - In master, the texture of the right and left cubes should be scaled differently from each other (one has the design "zoomed in"), and the center cube should look zoomed out.
    - In this PR, the cube textures should be as they appear in the provided auto test. i.e. the cubes on the left and right should have textures "zoomed out," while the cube in the center gives a close-up on the texture
    - **Note:** The automated test might be rather sensitive to changes in ambient lighting, but we only care about if the design has the correct offset/scale.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16349/HighFidelity-Beta-PR14679-f349e48.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16349/HighFidelity-Beta-PR14679-f349e48.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16349/HighFidelity-Beta-Interface-PR14679-f349e48.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16349/HighFidelity-Beta-PR14679-f349e48.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16349/HighFidelity-Beta-Interface-PR14679-f349e48.exe

**Comment by sabrina-shanman** [comment]
Jimi helped me acquire some models to help reproduce the bug in the Manuscript case, and it seems this PR does NOT fix the bug. So, this PR in its current state should not close the case.

**Comment by sabrina-shanman** [comment]
I'm hesitant to push this PR forward. It's a small change but it's unclear if it will do anything. I'll have to think it over as I'm working on a fix for the real repro.

**Comment by sabrina-shanman** [comment]
ModelUVScaling and ModelUVTranslation are showing up in the repro files in the places this PR uses, so I'm now reasonably sure this PR is a step in the right direction for UV scaling, even though it doesn't solve the issue specific to this Manuscript case.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16413/HighFidelity-Beta-PR14679-00a9e9e.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16429/HighFidelity-Beta-PR14679-5173577.apk

**Comment by sabrina-shanman** [comment]
It looks like the compiler warnings caught a bug.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16429/HighFidelity-Beta-PR14679-5173577.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16429/HighFidelity-Beta-Interface-PR14679-5173577.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16429/HighFidelity-Beta-PR14679-5173577.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16429/HighFidelity-Beta-Interface-PR14679-5173577.exe

File: libraries/fbx/src/FBXSerializer_Material.cpp [hunk 1]
```diff
@@ -59,13 +59,18 @@ HFMTexture FBXSerializer::getTexture(const QString& textureID) {
...
-        texture.transform.setScale(scaling);
+        texture.transform.postScale(scaling);
 
         if ((p.UVSet != "map1") && (p.UVSet != "UVSet0")) {
             texture.texcoordSet = 1;
         }
         texture.texcoordSetName = p.UVSet;
     }
+    if (_materialParams.contains(materialID)) {
+        auto materialParam = _materialParams[materialID];
```
Comment (SamGondelman) [inline libraries/fbx/src/FBXSerializer_Material.cpp]: This could be:
```
auto materialParamItr = _materialParams.find(materialID);
if (materialParamItr != _materialParams.end()) {
        auto materialParam = materialParamItr.value();
```

**Comment by sabrina-shanman** [comment]
I addressed your small change requests, and made the reasoning behind property lookup more clear. Also, after a look at my test file, it seems there is still an additional empty array element for properties which use the 3d vector, so I don't think there is a significant chance there is a 3d version of those scale/translation properties. I've decided to remove them in favor of simplicity.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16450/HighFidelity-Beta-PR14679-df17614.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16450/HighFidelity-Beta-Interface-PR14679-df17614.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16450/HighFidelity-Beta-PR14679-df17614.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16450/HighFidelity-Beta-Interface-PR14679-df17614.exe

**Comment by qligmaloney** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16752/HighFidelity-Beta-PR14679-df17614.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16752/HighFidelity-Beta-PR14679-df17614.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16752/HighFidelity-Beta-Interface-PR14679-df17614.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16752/HighFidelity-Beta-PR14679-df17614.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16752/HighFidelity-Beta-Interface-PR14679-df17614.exe

**Comment by qliejahner** [comment]
Currently testing
